homeassistant/components/bluetooth/passive_update_processor.py

```diff
@@ -35,22 +46,149 @@ class PassiveBluetoothEntityKey:
 
 _T = TypeVar("_T")
 
+STORAGE_KEY = "bluetooth.passive_update_processor"
+PASSIVE_UPDATE_PROCESSOR = "passive_update_processor"
+
+
+@dataclasses.dataclass(slots=True, frozen=False)
+class PassiveBluetoothProcessorData:
+    """Data for the passive bluetooth processor."""
+
+    coordinators: set[PassiveBluetoothProcessorCoordinator] = dataclasses.field(
+        default_factory=set
+    )
+    restore_data: dict[str, dict[str, dict[str, Any]]] = dataclasses.field(
+        default_factory=dict
+    )
+
+
+class PassiveBluetoothEntityKeyDict(TypedDict):
+    """Passive bluetooth entity key dict."""
+
+    key: str
+    device_id: str | None
+
+
+class RestoredData(TypedDict):
+    """Restored data."""
+
+    devices: dict[str | None, DeviceInfo]
+    entity_descriptions: dict[PassiveBluetoothEntityKeyDict, dict[str, Any]]
+    entity_names: dict[PassiveBluetoothEntityKeyDict, str | None]
+    entity_data: dict[PassiveBluetoothEntityKeyDict, Any]
+
 
 @dataclasses.dataclass(slots=True, frozen=True)
 class PassiveBluetoothDataUpdate(Generic[_T]):
     """Generic bluetooth data."""
 
     devices: dict[str | None, DeviceInfo] = dataclasses.field(default_factory=dict)
-    entity_descriptions: Mapping[
+    entity_descriptions: dict[
         PassiveBluetoothEntityKey, EntityDescription
     ] = dataclasses.field(default_factory=dict)
-    entity_names: Mapping[PassiveBluetoothEntityKey, str | None] = dataclasses.field(
+    entity_names: dict[PassiveBluetoothEntityKey, str | None] = dataclasses.field(
         default_factory=dict
     )
-    entity_data: Mapping[PassiveBluetoothEntityKey, _T] = dataclasses.field(
+    entity_data: dict[PassiveBluetoothEntityKey, _T] = dataclasses.field(
         default_factory=dict
     )
 
+    def update(self, new_data: PassiveBluetoothDataUpdate[_T]) -> None:
+        """Update the data."""
+        self.devices.update(new_data.devices)
+        self.entity_descriptions.update(new_data.entity_descriptions)
+        self.entity_data.update(new_data.entity_data)
+        self.entity_names.update(new_data.entity_names)
+
+    def async_restore_data(self) -> dict[str, Any]:
+        """Serialize restore data to storage."""
+        return {
+            "devices": self.devices,
+            "entity_descriptions": self.entity_descriptions,
+            "entity_names": self.entity_names,
+            "entity_data": self.entity_data,
+        }
+
+    @classmethod
+    def from_storage(cls, restored_data: RestoredData) -> PassiveBluetoothDataUpdate:
+        """Restore data from storage."""
+        return cls(
+            devices=restored_data["devices"],
+            entity_descriptions={
+                PassiveBluetoothEntityKey(
+                    **passive_bluetooth_entity_key
+                ): EntityDescription(**description)
```

Comment:
This is the wrong entity description

Reply:
When we register the processor we should have a new function to register with restore that passes the entity description class and optional restore key which defaults to current platform